src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts

```diff
@@ -410,6 +410,49 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
 		}
 		return value;
 	}
+
+	async createProfileFromShellAndShellArgs(shell?: unknown, shellArgs?: unknown): Promise<ITerminalProfile | undefined> {
+		const detectedProfile = this._terminalService.availableProfiles?.find(p => p.path === shell);
+		const fallbackProfile = (await this.getDefaultProfile({
+			remoteAuthority: this._remoteAgentService.getConnection()?.remoteAuthority,
+			os: this._primaryBackendOs!
+		}));
+		const profile = detectedProfile || fallbackProfile;
+		const args = typeof shellArgs === 'string' || Array.isArray(shellArgs) ? shellArgs : profile.args;
```

Comment:
Can use `_isValidShellArgs` here which also makes sure string is only used on Windows.